### ISSUE TITLE
Xcode 8 and swift 2.3 - Use of deprecated rand() function in RxCocoa Driver.swift

### ISSUE THREAD

**Issue by stulevine** [ReactiveX/RxSwift#886]
:warning: If you don't have something to report in the following format, it will probably be easier and faster to ask in the [slack channel](http://http://slack.rxswift.org/) first. :warning: 

**Short description of the issue**:

  Our iOS Project uses RxSwift (via Pod install).  With the recent release of Xcode 8, and the pending adoption of iOS 10, I decided to convert our Xcode 7 project to Xcode 8 and Swift 2.3.  However, after the conversion, and when Build the RxSwift framework, I discovered a fatal error in the RxCocoa Driver class (Driver.h).  Apparently, it's still using the rand() function, which is no longer supported in Swift 2.3.  The recommendation is to use arc4random().  And, while I did make local changes to quell this error message, it would be better to perhaps have the change implemented by a contributor.  Having said this, should I fork and create a pull request with this single line change?

**Self contained code example that reproduces the issue**:
In the RxCocoa framework, at line# 214 in Driver.swift, the following code will fail to build unless rand() is replaced with arc4random()
```swift
func _forceCompilerToStopDoingInsaneOptimizationsThatBreakCode(scheduler: SchedulerType) {
    let a: Int32 = 1
    let b = 314 + Int32(rand() & 1)
    if a == b {
        print(scheduler)
    }
}
  
// If we can't get a self contained code example that reproduces the issue, there is a big chance we won't be able
// to help you because there is not much we can do.
// `Self contained code example` means:
// * that we should be able to just run the provided code without changing it.
// * that it will reproduce the issue upon running
```

**Xcode version**:

```
  _8.0_
```

**Expected outcome**:

  _RxSwift should build without errors in Xcode 8 with swift 2.3_

**What actually happens**:

  _Xcode reports an error stating that rand() function is not supported in Swift 2.3 and that we shuold use arc4random() instead_

:warning: Fields below are optional for general issues or in case those questions aren't related to your issue, but filling them out will increase the chances of getting your issue resolved. :warning:

**Installation method**:
  - [ X] CocoaPods
  - [ ] Carthage
  - [ ] Git submodules

**I have multiple versions of Xcode installed**:
  (so we can know if this is a potential cause of your issue)
  - [x ] yes (7.3.1 and 8.0 - although this issue is specific to Xcode 8, iOS 10 and Swift 2.3)
  - [ ] no

**Level of RxSwift knowledge**:
  (this is so we can understand your level of knowledge
    and formulate the response in an appropriate manner)
  - [ ] just starting
  - [ ] I have a small code base
  - [X ] I have a significant code base

**Comment by kzaher** [comment]
Hi @stulevine ,

I think this is already fixed in 2.6.0 release for Swift 2.0. Can you please provide me with information which release are you using?

**Comment by stulevine** [comment]
Oh snap! Thanks @kzaher. Good catch.  My Podfile was configured for only patch level updates.  Re-configured for minor updates and was able to update RxSwift and RxCocoa to v2.6.
